pulsar-testclient/src/main/java/org/apache/pulsar/testclient/CmdGenerateDocumentation.java

```diff
@@ -24,57 +24,46 @@
-public class CmdGenerateDocumentation {
+@Command(description = "Generate documentation automatically.", showDefaultValues = true, scope = ScopeType.INHERIT)
+public class CmdGenerateDocumentation extends CmdBase{
 
-    @Command(description = "Generate documentation automatically.", showDefaultValues = true, scope = ScopeType.INHERIT)
-    static class Arguments {
 
-        @Option(names = {"-h", "--help"}, description = "Help message", help = true)
-        boolean help;
+    @Option(names = {"-h", "--help"}, description = "Help message", help = true)
+    boolean help;
 
-        @Option(names = {"-n", "--command-names"}, description = "List of command names")
-        private List<String> commandNames = new ArrayList<>();
+    @Option(names = {"-n", "--command-names"}, description = "List of command names")
+    private List<String> commandNames = new ArrayList<>();
 
+    public CmdGenerateDocumentation() {
+        super("gen-doc");
     }
 
-    public static void main(String[] args) throws Exception {
-        final Arguments arguments = new Arguments();
-        CommandLine commander = new CommandLine(arguments);
-        commander.setCommandName("pulsar-perf gen-doc");
-        try {
-            commander.parseArgs(args);
-        } catch (ParameterException e) {
-            System.out.println(e.getMessage());
-            commander.usage(commander.getOut());
-            PerfClientUtils.exit(1);
-        }
-
 
-        if (arguments.help) {
+    @Override
+    public void run() throws Exception {
+        CommandLine commander = super.getCommander();
+        if (this.help) {
             commander.usage(commander.getOut());
             PerfClientUtils.exit(1);
         }
 
         Map<String, Class<?>> cmdClassMap = new LinkedHashMap<>();
-        cmdClassMap.put("produce", Class.forName("org.apache.pulsar.testclient.PerformanceProducer$Arguments"));
-        cmdClassMap.put("consume", Class.forName("org.apache.pulsar.testclient.PerformanceConsumer$Arguments"));
-        cmdClassMap.put("transaction", Class.forName("org.apache.pulsar.testclient.PerformanceTransaction$Arguments"));
-        cmdClassMap.put("read", Class.forName("org.apache.pulsar.testclient.PerformanceReader$Arguments"));
-        cmdClassMap.put("monitor-brokers", Class.forName("org.apache.pulsar.testclient.BrokerMonitor$Arguments"));
-        cmdClassMap.put("simulation-client",
-                Class.forName("org.apache.pulsar.testclient.LoadSimulationClient$MainArguments"));
-        cmdClassMap.put("simulation-controller",
-                Class.forName("org.apache.pulsar.testclient.LoadSimulationController$MainArguments"));
-        cmdClassMap.put("websocket-producer",
-                Class.forName("org.apache.pulsar.proxy.socket.client.PerformanceClient$Arguments"));
-        cmdClassMap.put("managed-ledger", Class.forName("org.apache.pulsar.testclient.ManagedLedgerWriter$Arguments"));
+        cmdClassMap.put("produce", PerformanceProducer.class);
```

Comment:
I remember we have multiple gen-doc class, you can improve here.

Reply:
https://github.com/apache/pulsar/blob/434ec1b884b26af6c3658df3a7b90c432e464973/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdGenerateDocument.java#L43-L46

You are right; I found another one.

But they are in different modules, and I think it's a bit difficult to merge them into one.